### PR TITLE
Move version out of LedgerMetadata

### PR DESCRIPTION
**Pull request by ivankelly**
The version is a property of the metadata store and not of the
LedgerMetadata object itself. Putting it in the LedgerMetadata forces
that LedgerMetadata to be mutable, as it needs to be updated before
being sent to the LedgerManager.

This change moves version out of LedgerMetadata and modifies the
LedgerManager to use Versioned<LedgerMetadata>.

This is a large change, but almost purely mechanical.

TestHttpService has been modified to remove a preexisting flake in a
test which had to be modified for this change.

Master issue: #281

### CODE REVIEW

**Comment by ivankelly** [comment]
I didn't do it in this change, but I want to change LedgerManager to use CompletableFutures instead of callbacks, because everywhere that uses it is converting to Futures anyhow.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java [hunk 1]
```diff
@@ -190,7 +190,9 @@
         // password, so that the bookie can avoid processing the keys for each entry
         this.ledgerKey = DigestManager.generateMasterKey(password);
         distributionSchedule = new RoundRobinDistributionSchedule(
-                metadata.getWriteQuorumSize(), metadata.getAckQuorumSize(), metadata.getEnsembleSize());
+                metadata.getValue().getWriteQuorumSize(),
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java]: Nit: call getValue() only once?

Comment (ivankelly) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java]: done

**Comment by ivankelly** [comment]
rerun bookie replication tests

**Comment by ivankelly** [comment]
rerun bookkeeper-server replication tests

**Comment by ivankelly** [comment]
rerun pr validation
rerun bookkeeper-server client tests

**Pull request merged**